sound/soc/intel/boards/Kconfig

```diff
@@ -483,6 +471,20 @@ config SND_SOC_INTEL_SOF_RT5682_MACH
 	   with rt5682 codec.
 	   Say Y if you have such a device.
 	   If unsure select "N".
+
+config SND_SOC_INTEL_SOF_PCM512x_MACH
+	tristate "SOF with TI PCM512x codec"
+	depends on I2C && ACPI
+	depends on (SND_SOC_SOF_HDA_LINK && (MFD_INTEL_LPSS || COMPILE_TEST)) ||\
```

Comment:
@kv2019i this should be HDA_AUDIO_CODEC. no? we don't support the plain vanilla hdac_hdmi stuff so probably need to restrict more here.

Reply:
@plbossart there's a
```
select SND_HDA_CODEC_HDMI if SND_SOC_SOF_HDA_AUDIO_CODEC
```
below, that's also what other drivers do. You're proposing
```
	depends on (SND_SOC_SOF_HDA_AUDIO_CODEC && (MFD_INTEL_LPSS || COMPILE_TEST)) ||\
		   (SND_SOC_SOF_BAYTRAIL && (X86_INTEL_LPSS || COMPILE_TEST))
	select SND_SOC_PCM512x_I2C
	select SND_HDA_CODEC_HDMI if SND_SOC_SOF_HDA_AUDIO_CODEC
```
?

Reply:
yes, the snippet above is what I meant. The conditional select if needed for the BAYTRAIL case.

Reply:
@plbossart @lyakh  Ack, this looks good.